### PR TITLE
DOC: clean-up of 0.24.2 whatsnew file

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Review by TomAugspurger — APPROVED** [review]

**Comment by jreback** [comment]
a more general comment. I find the 'Fixed Regressions' section way harder to read than our separated Bug Fixes section (e.g. by area). Ideally should either reorg that and have sub-topics, or just combine them to one (maybe have a way of flagging a regression vs a bug).

**Comment by TomAugspurger** [comment]
We could use an HTML label like scikit-learn does in
https://scikit-learn.org/stable/whats_new.html.

**Comment by jorisvandenbossche** [comment]
I personally like the separate regressions section, but they indeed could also be subdivided (the labels of sklearn are also nice, but they group everything (also enhancements) per topic). 
But given that this PR was only cleaning up the whitespace for what is already there, going to leave that discussion for another time.
